### PR TITLE
Clarify which packet losses may be ignored

### LINKED ISSUE #3456: Clarify when it is ok to ignore packet loss
Currently, the "Ignoring Loss of Undecryptable Packets" section indicates that some loss can be ignored, but now that we have multiple packet number spaces, I think we can constrain the set of ignorable losses to those below the smallest acknowledged packet in a packet number space.

Ignoring the loss of a subsequently sent packet was not the intent.

### PR DESCRIPTION
**Pull request by ianswett**
Fixes #3456

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -690,12 +690,14 @@ losses or increases in the ECN-CE counter.
...
-During the handshake, some packet protection keys might not be	
-available when a packet arrives. In particular, Handshake and 0-RTT packets	
-cannot be processed until the Initial packets arrive, and 1-RTT packets	
-cannot be processed until the handshake completes.  Endpoints MAY	
-ignore the loss of Handshake, 0-RTT, and 1-RTT packets that might arrive before	
-the peer has packet protection keys to process those packets.	
+During the handshake, some packet protection keys might not be available when
+a packet arrives and the receiver can choose to drop the packet. In particular,
+Handshake and 0-RTT packets cannot be processed until the Initial packets
+arrive, and 1-RTT packets	cannot be processed until the handshake completes.
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md]: An errant tab here.
```suggestion
arrive and 1-RTT packets cannot be processed until the handshake completes.
```

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -690,12 +690,14 @@ losses or increases in the ECN-CE counter.
...
+Endpoints MAY	ignore the loss of Handshake, 0-RTT, and 1-RTT packets that might
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md]: and here
```suggestion
Endpoints MAY ignore the loss of Handshake, 0-RTT, and 1-RTT packets that might
```

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -690,12 +690,14 @@ losses or increases in the ECN-CE counter.
...
-ignore the loss of Handshake, 0-RTT, and 1-RTT packets that might arrive before	
-the peer has packet protection keys to process those packets.	
+During the handshake, some packet protection keys might not be available when
+a packet arrives and the receiver can choose to drop the packet. In particular,
+Handshake and 0-RTT packets cannot be processed until the Initial packets
+arrive and 1-RTT packets cannot be processed until the handshake completes.
+Endpoints MAY ignore the loss of Handshake, 0-RTT, and 1-RTT packets that might
+have arrived before the peer has packet protection keys to process those
+packets. Endpoints MUST NOT ignore the loss of packets that were sent after
+the earliest acknowledged packet in a given packet number space.
```
Comment (mikkelfj) [inline draft-ietf-quic-recovery.md]: How would that (not) impact the congestion controller?

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: I'm unsure what you're asking?

Comment (mikkelfj) [reply draft-ietf-quic-recovery.md]: If you send a packet you consume bytes in flight. If you do not get an ack for that packet, eventually it is lost. I'm not sure what it means to ignore in that context unless you complete ignore an earlier packet space. But perhaps the text refers to ignoring received packets, not sent packets. In that case I would say drop, not ignore.

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: #3415 makes that extra clear I think.

**Pull request merged**